### PR TITLE
[HZ-1388] Fix keyed idle message handling

### PR DESCRIPTION
**Pull request by Fly-Style**
Fixes the behavior according to the TDD: 
https://github.com/hazelcast/hazelcast/blob/021e2af77ad6ba6d32d56a623fbe43f30fd76864/docs/design/sql/14-keyed-watermark-support.md#handling-of-keyed-idle-messages

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/jet/impl/execution/KeyedWatermarkCoalescerTest.java [hunk 1]
```diff
@@ -20,22 +20,22 @@
...
+import static com.hazelcast.jet.impl.execution.WatermarkCoalescer.IDLE_MESSAGE;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singleton;
 import static java.util.Collections.singletonList;
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class KeyedWatermarkCoalescerTest extends JetTestSupport {
```
Comment (viliam-durina) [inline hazelcast/src/test/java/com/hazelcast/jet/impl/execution/KeyedWatermarkCoalescerTest.java:36]: Failing tests here.

**Review by viliam-durina — CHANGES_REQUESTED** [review]

**Comment by Fly-Style** [comment]
run-lab-run

**Comment by Fly-Style** [comment]
run-lab-run

**Comment by frant-hartm** [comment]
run-ee-compile

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**